designs/2019-allow-all-directives-in-line-comments/README.md

```diff
@@ -0,0 +1,112 @@
+- Start Date: 2019-07-22
+- RFC PR: (leave this empty, to be filled in later)
+- Authors: Jordan Eldredge <jordan@jordaneldredge.com>
+
+# Allow All Directives in Line Comments
+
+## Summary
+
+ESLint currently only allows the following types in directives to be used inside of block comments (`/* Block Comment */`):
+
+* `exported`
+* `global(s)`
+* `eslint-(en|dis)able`
+* `eslint`
+
+Currently if a user adds one of these directives in a line comment (`// Line Comment`) the directive will be silently ignored. Which can be very confusing.
+
+I propose that we allow all directives to work in either type of comment.
+
+## Motivation
+
+As a user of ESLint I occasionally wish to disable a rule, declare a valid global or add some other file level config. In doing so I often forget that these directives are only valid inside block level comments and add them as inline comments by mistake. When the comment fails to have its desired effect I consult the documentation to see what I did wrong. Did I remember the directive syntax or name incorrectly? On more than one occasion the problem has been that I used the wrong kind of comment.
+
+It would be nice if I didn't have to think about the comment type that I'm using.
+
+## Detailed Design
+
+The function `getDirectiveComments` in [`linter.js`](https://github.com/eslint/eslint/blob/1fb362093a65b99456a11029967d9ee0c31fd697/lib/linter/linter.js#L263) currently does an `if (comment.type === "Block")` check before it looks for some types of directives.
+
+I believe we could simply remove that check.
+
+For `/* eslint-env */` directives which, by necessity, are parsed out of the file before an AST is avaliable, the regular expression in [linter.js](https://github.com/eslint/eslint/blob/fb08b7c9d28bc68864eb940e26df274059228b6a/lib/linter/linter.js#L406) would need to be updated to allow it to match line comments.
+
+## Documentation
+
+We would need to update the [configuration](https://eslint.org/docs/user-guide/configuring#using-configuration-comments) documentation to remove the places where it states that some types of directives must use block comments. We would replace each of those comments with a note that "Before version _x.x_ this directive only worked in block comments" so that users using an older version of ESLint could figure out why their line comments were ignored.
+
+## Drawbacks
+
+I don't see any drawbacks to this change.
+
+## Backwards Compatibility Analysis
+
```

Comment:
Can you update this section with the explanation that those changes will be backwards-compatible but would be considered breaking due to the example with `// global` comment.